crates/sui-graphql-rpc/src/server/sui_sdk_data_provider.rs

```diff
@@ -364,6 +364,10 @@ pub(crate) async fn convert_to_epoch(cl: &SuiClient, gcs: &NativeGasCostSummary)
             gas_summary: Some(gas_summary),
         }),
         protocol_configs: Some(protocol_configs),
+        start_timestamp: Some(DateTime::from(system_state.epoch_start_timestamp_ms)),
+        end_timestamp: Some(DateTime::from(
+            system_state.epoch_start_timestamp_ms + system_state.epoch_duration_ms,
+        )),
```

Comment:
This isn't going to be accurate and instead would need the start time of the next epoch in order to calculate this. I wonder if its even useful to have 'end_timestamp' given that. 

The reason why this isn't exactly accurate is that the validators wait till `epoch_duration_ms` has elapsed and then they begin the process to close out the epoch. This process could take a number of sections to accomplish, drifting slightly longer than the `epoch_duration_ms` value

Reply:
The correct place to get this information from is `EndOfEpochInfo`:

https://github.com/MystenLabs/sui/blob/868bbcc46c66a01590e2b849f8325b294a5bda39/crates/sui-json-rpc-types/src/sui_extended.rs#L54

And if we can't get it (i.e. because the epoch hasn't ended yet), then we emit `None`.

Reply:
Awesome, thanks @bmwill and @amnn

Reply:
So what do we want to do with this field? I guess it depends if we want to drop the burden of computing this to whoever service/user needs it.